.NET/Microsoft.Recognizers.Text.DateTime/Parsers/CJK/BaseCJKDatePeriodParser.cs

```diff
@@ -1033,19 +1108,59 @@ private DateTimeResolutionResult ParseOneWordPeriod(string text, DateObject refe
                         ret.Success = true;
                         return ret;
                     }
+                    else if (!string.IsNullOrEmpty(match.Groups["restof"].Value))
+                    {
+                        var durationStr = match.Groups["duration"].Value;
+                        var durationUnit = this.config.UnitMap[durationStr];
+                        DateObject beginDate;
+                        DateObject endDate = beginDate = referenceDate;
+                        var durationTimex = string.Empty;
+
+                        switch (durationUnit)
+                        {
+                            case Constants.TimexWeek:
+                                var diff = Constants.WeekDayCount - (beginDate.DayOfWeek == 0 ? Constants.WeekDayCount : (int)beginDate.DayOfWeek);
+                                endDate = beginDate.AddDays(diff);
+                                durationTimex = "P" + diff + Constants.TimexDay;
+                                break;
+
+                            case Constants.TimexMonthFull:
+                                endDate = DateObject.MinValue.SafeCreateFromValue(beginDate.Year, beginDate.Month, 1);
+                                endDate = endDate.AddMonths(1).AddDays(-1);
+                                diff = endDate.Day - beginDate.Day + 1;
+                                durationTimex = "P" + diff + Constants.TimexDay;
+                                break;
+
+                            case Constants.TimexYear:
+                                endDate = DateObject.MinValue.SafeCreateFromValue(beginDate.Year, 12, 1);
+                                endDate = endDate.AddMonths(1).AddDays(-1);
+                                diff = endDate.DayOfYear - beginDate.DayOfYear + 1;
```

Comment:
Diff and timex to a timex gen method instead of repeating string concat everywhere.

Reply:
Actually, the whole timex in 1142 too.

Reply:
replaced with GenerateDatePeriodTimexWithDiff

Reply:
If the diff is no used here, why not calc inside the timex gen method?

Reply:
it is used in the "TimexWeek" case to calculate the endDate. 
I have moved all the switch in the timex method now.